app/jobs/recalculate_handover_date_job.rb

```diff
@@ -5,16 +5,35 @@ class RecalculateHandoverDateJob < ApplicationJob
 
   def perform(nomis_offender_id)
     offender = OffenderService.get_offender(nomis_offender_id)
-    return if offender.nil? || offender.sentenced? == false
-
-    begin
+    if offender&.inside_omic_policy?
       # Recalculate handover dates, which will trigger a push to the Community API after_save
-      CalculatedHandoverDate.recalculate_for(offender)
-    rescue Faraday::ClientError # rubocop:disable Lint/SuppressedException
-      # Swallow Community API errors to allow this job to complete successfully.
-      # The calculated handover date will not be saved, and we'll retry again in tomorrow night's cron.
-      # Without this, bad data in nDelius will cause our retry queue to continually grow as fresh jobs
-      # are queued every night alongside old failed jobs continually retrying.
+      recalculate_dates_for(offender)
+    end
+  end
+
+private
+
+  def recalculate_dates_for(offender)
+    # we have to go direct to handover data to avoid being blocked when COM is responsible
+    handover = HandoverDateService.handover(offender)
+    case_info = CaseInformation.find_by!(nomis_offender_id: offender.offender_no)
+    record = case_info.calculated_handover_date.presence || case_info.build_calculated_handover_date
+    record.assign_attributes(
+      start_date: handover.start_date,
+      handover_date: handover.handover_date,
+      reason: handover.reason
+    )
+    record.save! if record.changed?
+    push_to_delius record
+  end
+
+  def push_to_delius record
+    # Don't push if the CaseInformation record is a manual entry (meaning it didn't match against nDelius)
+    # This avoids 404 Not Found errors for offenders who don't exist in nDelius (they could be Scottish, etc.)
+    # Don't push if the dates haven't changed
+    if !record.case_information.manual_entry &&
+      record.saved_change_to_start_date? || record.saved_change_to_handover_date?
```

Comment:
I think you might need to add brackets in here:

```suggestion
    if !record.case_information.manual_entry &&
      ( record.saved_change_to_start_date? || record.saved_change_to_handover_date? )
```

Without them, I think it could incorrectly push handover dates on manual entries.

According to `irb`:

```ruby
false && true || true
# => true

false && ( true || true )
# => false
```

Looks like this combination isn't covered by any tests..?